src/vec.rs

```diff
@@ -859,25 +899,75 @@ impl<T, const N: usize> Vec<T, N> {
         }
 
         // Stage 1: Nothing was deleted.
-        process_loop::<F, T, N, false>(original_len, &mut f, &mut g);
+        process_loop::<F, T, N, A, false>(original_len, &mut f, &mut g);
 
         // Stage 2: Some elements were deleted.
-        process_loop::<F, T, N, true>(original_len, &mut f, &mut g);
+        process_loop::<F, T, N, A, true>(original_len, &mut f, &mut g);
 
         // All item are processed. This can be optimized to `set_len` by LLVM.
         drop(g);
     }
 }
 
+impl<const N: usize, A> Vec<u8, N, A> {
+    /// Transmutes the filled buffer to the output type. The storage of the length of the [`Vec`] does
+    /// not participate in the transmutation.
+    ///
+    /// # Safety
+    ///
+    /// - The buffer must be full.
+    /// - The size of the output type must be equal to the size of the **buffer** (rather than [`Vec`]).
+    /// - The alignment of the `Vec` must be equal or be a multiple of the alignment of the output.
+    ///
+    /// # Panics
+    ///
+    /// In debug mode, this method panics...
+    ///
+    /// - If the buffer isn't full.
+    /// - If the size of the output type doesn't match the size of the buffer.
+    /// - If the alignment of the `Vec` isn't equal or a multiple of the alignment of the output.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use heapless::Vec;
+    ///
+    /// let mut v: Vec<u8, 2, u16> = Vec::new();
+    /// v.extend_from_slice(&[0, 0]).unwrap();
+    /// let number: u16 = unsafe { v.transmute_buffer() };
+    /// assert_eq!(number, 0u16);
+    /// ```
```

Comment:
I'd like to see a more elaborate example here. While the example shows how to use this method, it is a bit too trivial and doesn't give me a good idea why this is needed in practice. Especially the fact that the `Vec` needs to be full feels quite limiting. In the example, you could transmute to `[MaybeUninit<u16>; 1]` without it being full. Of course you wouldn't be able to read the partially uninitialized value, but with a longer buffer it may make sense.

I think ideally, we would have a safe method that returns `[MaybeUninit<A>; M]`, but there currently doesn't seem a way to calculate or even assert the size of `M` with const generics.